### PR TITLE
[Concept Entry] Rust loops #3374

### PR DESCRIPTION
**Pull request by moha230**
### Description
- added concept entry rust loops
- added a folder loops
- added a file loops.md
- Closes issue #3374

### Type of Change
- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by moha230** [comment]
Hello @SSwiniarski I have done the entry . Thank you

**Comment by moha230** [comment]
hello @SSwiniarski  i have an issue in the [Concept Entry] Rust loops #3374 #3856 I don't know why its like this please help

File: content/rust/concepts/loops/loops.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Loops'
+CatalogContent:
+  - 'Learn-rust'
+  - 'paths/computer-science'
+---
+
+A `loop` is a control structure that allows you to execute a block of code repeatedly. Loops are used to perform tasks that need to be done multiple times, such as iterating through a list of items, performing calculations, or waiting for a specific condition to be met.There typically three main types `for`, `while` and `loop`.
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:13]:
```suggestion
A **`loop`** is a control structure that enables the repetitive execution of a block of code. `Loops` are employed for tasks that require multiple iterations, such as iterating through a list of items, performing calculations, or waiting for a specific condition to be met. There are typically three main types: `for`, `while`, and `loop`.
```
we don't prefer to use pronouns. Refrain form using them.

File: content/rust/concepts/loops/loops.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+Title: 'Loops'
+Description: 'A loop is a control structure that allows you to repeatedly execute a block of code.'
+Subjects:
+  - 'Computer Science'
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:5]:
```suggestion
  - 'Computer Science'
  - 'Code Foundations'
```

File: content/rust/concepts/loops/loops.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+Tags:
+  - 'Loops'
+CatalogContent:
+  - 'Learn-rust'
+  - 'paths/computer-science'
+---
+
+A `loop` is a control structure that allows you to execute a block of code repeatedly. Loops are used to perform tasks that need to be done multiple times, such as iterating through a list of items, performing calculations, or waiting for a specific condition to be met.There typically three main types `for`, `while` and `loop`.
+
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:14]:
```suggestion
## Syntax

```
We always add syntax heading then we add the example heading. We can make the different kinds of loops as sub headings.

File: content/rust/concepts/loops/loops.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+## for loops
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:15]:
```suggestion
### For loop
```

File: content/rust/concepts/loops/loops.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+The `for` loop is used to iterate over collections or ranges.It automatically iterates through elements and executes a block of code for each item in the collection or range.It's commonly used for iterating over arrays, vectors, and other iterable data structures.
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:17]:
```suggestion
The `for loop` is used to iterate over collections or ranges. It automatically iterates through elements and executes a code block for each item in the pool or scope. It's commonly used for iterating over `arrays`, `vectors`, and other iterable data structures.
```

File: content/rust/concepts/loops/loops.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+```rust
+let numbers = ["A", 2,20, 4, 5];
+for num in numbers.iter() {
+    // Code to be executed for each element in the array
+    println!("Number: {}", num);
+}
+```
+
+## while loops
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:27]:
```suggestion
### While loop
```

File: content/rust/concepts/loops/loops.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+```rust
+let mut counter = 0;
+while counter < 5 {
+    // Code to be executed while the condition is true
+    counter += 1;
+}
+```
+
+## loop loops
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:39]:
```suggestion
### Loop 
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have few comments work on them. Once you commit these I will do another check.

File: content/rust/concepts/loops/loops.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        if input.trim() == "exit" {
+            break;
+        }
+    }
+}
+```
+
+## Example
+
+This example demonstrates a `while` loop that iterates from `5` to `10` as the counter variable starts from `5`.The second example shows a `for` loop that iterates over an array that containe numbers `1` to `5` .
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:59]:
```suggestion
This example demonstrates a `while loop` that iterates from `5` to `10` as the counter variable starts from `5`. The second example shows a `for` loop that iterates over an `array` that contains numbers `1` to `5`.
```

File: content/rust/concepts/loops/loops.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+let mut counter = 0;
+while counter < 5 {
+    // Code to be executed while the condition is true
+    counter += 1;
+}
+```
+
+## loop loops
+
+The `loop` keyword creates an infinite loop that continues executing the code block indefinitely until it's explicitly interrupted using the break statement.It's useful when you need to perform a task until a certain condition is met within the loop.
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:41]:
```suggestion
The `loop` keyword creates an infinite loop that continues executing the code block indefinitely until it's explicitly interrupted using the break statement. It's useful when you need to perform a task until a certain condition is met within the loop.
```

File: content/rust/concepts/loops/loops.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+let numbers = ["A", 2,20, 4, 5];
+for num in numbers.iter() {
+    // Code to be executed for each element in the array
+    println!("Number: {}", num);
+}
+```
+
+## while loops
+
+The `while` loop repeatedly executes a block of code as long as a specified condition remains true.It checks the condition before entering the loop, and if the condition is initially false, the loop won't run.While loops are useful when you don't know in advance how many iterations are needed.
```
Comment (dakshdeepHERE) [inline content/rust/concepts/loops/loops.md:29]:
```suggestion
The `while loop` repeatedly executes a block of code as long as a specified condition remains true. It checks the condition before entering the loop, and if the condition is initially `false`, the loop won't run. `While loops` are useful when you don't know in advance how many iterations are needed.
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]

**Comment by moha230** [comment]
Hello @dakshdeepHERE its done . Thank you

**Review by dakshdeepHERE — APPROVED** [review]
Did some small changes. Overall look good for 2nd review.

File: content/rust/concepts/loops/loops.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        if input.trim() == "exit" {
+            break;
+        }
+    }
+}
+```
+
+## Example
+
+This example demonstrates a `while loop` that iterates from `5` to `10` as the counter variable starts from `5`. The second example shows a `for` loop that iterates over an `array` that contains numbers `1` to `5`.
```
Comment (caupolicandiaz) [inline content/rust/concepts/loops/loops.md:62]:
```suggestion
This example demonstrates a `while loop` that iterates through each integer from `5` to `10`. The second example uses a `for` loop to iterate over an array that contains the integers in the range `1` to `5`.
```

File: content/rust/concepts/loops/loops.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
+---
+Title: 'Loops'
+Description: 'A loop is a control structure that allows you to repeatedly execute a block of code.'
+Subjects:
+  - 'Computer Science'
+  - 'Code Foundations'
+Tags:
+  - 'Loops'
+CatalogContent:
+  - 'Learn-rust'
```
Comment (caupolicandiaz) [inline content/rust/concepts/loops/loops.md:10]:
```suggestion
  - 'rust-for-programmers'
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@moha230 I've added my initial comments. Please update the entry, and tag me when you've made the changes. Thanks

File: content/rust/concepts/loops/loops.md [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+```pseudo
+let mut counter = 0;
+while counter < 5 {
+    // Code to be executed while the condition is true
+    counter += 1;
+}
+```
+
+### Loop
```
[the remark this answers is not in the document]
Comment (moha230) [reply content/rust/concepts/loops/loops.md:42]: Hello @caupolicandiaz If the sub heading on line 42 changes to infinite loop should we also change the intro at line 14 ? . Suggested change :A **loop** is a control structure that enables the repetitive execution of a block of code. Loops are employed for tasks that require multiple iterations, such as iterating through a list of items, performing calculations, or waiting for a specific condition to be met. There are typically three main types: `for`, `while` and  infinite `loop`. Thank you

**Comment by moha230** [comment]
Hello @caupolicandiaz all done . Thank you

**Pull request merged**